main.js

```diff
@@ -93,115 +90,166 @@
+                record.incidentDay = parseInt(record.CREATED_ON.substring(8,10));
+            }
+        },
+
+        // Calls from the library db
+        "Library": {
+          id: "2ba0788a-2f35-43aa-a47c-89c75f55cf9d",
+          primaryFiltering: 'WHERE "Name" LIKE \'%OAKLAND%\'',
+          latLong: ['Lat', 'Lon'],
+          icon: iconTypes.LIBRARY_ICON,
+
+          title: (record) => record['Name'],
+          popup: (record) => `
+            <strong>${record.Name}</strong>
+            <br> Address: ${record.Address}
+            <br> Phone: ${record.Phone}
+            <br> Monday: ${record.MoOpen.substring(0, 5)} - ${record.MoClose.substring(0, 5)}
+            <br> Tuesday: ${record.TuOpen.substring(0, 5)} - ${record.TuClose.substring(0, 5)}
+            <br> Wednesday: ${record.WeOpen.substring(0, 5)} - ${record.WeClose.substring(0, 5)}
+            <br> Thursday: ${record.ThOpen.substring(0, 5)} - ${record.ThClose.substring(0, 5)}
+            <br> Friday: ${record.FrOpen.substring(0, 5)} - ${record.FrClose.substring(0, 5)}
+            <br> Saturday: ${record.SaOpen.substring(0, 5)} - ${record.SaClose.substring(0, 5)}
+            <br> Sunday: ${record.SuOpen.substring(0, 5)} - ${record.SuClose.substring(0, 5)}
+            `
+        }
+    };
+
+    const WPRDC_QUERY_PREFIX = 'SELECT * FROM "';
+    const WPRDC_QUERY_SUFFIX = '" ';
+
+    // Fetch data from West Pennsylvania Regional Data Center using the SQL API
+    function fetchWPRDCData(dataSourceName, options) {
+      if (!options) {
+        options = {};
+      }
+
+      console.group(`${dataSourceName} API`);
+
+      const dataSource = WPRDC_DATA_SOURCES[dataSourceName];
+      let query = WPRDC_QUERY_PREFIX + dataSource.id + WPRDC_QUERY_SUFFIX + dataSource.primaryFiltering;
+
+      if (options.limit) {
+        console.log(`Limit set to: ${options.limit}`);
+        query += ' LIMIT ' + options.limit;
+      }
+
+      console.log(`Final query: ${query}`);
+      console.groupEnd();
+
+      return fetch(WPRDC_BASE_URL + query)
+          // TODO: ensure 200 response
+          .then((response) => response.json())
+          // TODO: should have some generic error handling for data
+          .catch((err) => console.log(err))
+          .then((data) => {
+              const records = data.result.records;
+
+              records.forEach((record, i) => {
+                  // TODO: Check browser compatability for `instanceof Function`
+                  if (dataSource.processRecord instanceof Function) {
+                      dataSource.processRecord(record, i);
+                  }
+
+                  const latLong = dataSource.latLong.map((fieldName) => record[fieldName]);
+                  const latLongNoNulls = latLong.some((field) => !!field);
+                  if (latLongNoNulls) {
+                      let title = null;
+                      if (dataSource.title instanceof Function) {
```

Comment:
Let's assume that all datasets have a title and that it is a function.In fact, could you add this for the police and 311? It can be the field that you're using now for the display.